File: package/main/src/Crypto/decodeBase32ToString.ts
import { decodeBase32 } from "./decodeBase32";

/**
 * Decodes a Base32 string to a UTF-8 string
 * @param {string} input - Base32 encoded string
 * @returns {string} Decoded string
 * @example decodeBase32ToString("JBSWY3DP"); // "Hello"
 */
export const decodeBase32ToString = (input: string): string => {
  return new TextDecoder().decode(decodeBase32(input));
};

File: package/umt_python/src/crypto/decode_base32_to_string.py
from .decode_base32 import decode_base32


def decode_base32_to_string(input_data: str) -> str:
    """
    Decodes a Base32 string to a UTF-8 string.

    Args:
        input_data: Base32 encoded string

    Returns:
        Decoded string

    Example:
        >>> decode_base32_to_string("JBSWY3DP")
        'Hello'
    """
    return decode_base32(input_data).decode("utf-8")
